overlord/snapstate/backend.go

```diff
@@ -79,3 +92,74 @@ func (s *defaultBackend) Download(name, channel string, meter progress.Meter) (s
 
 	return downloadedSnapFile, snap.Developer(), nil
 }
+
+func (s *defaultBackend) CheckSnap(snapFilePath string, flags snappy.InstallFlags) error {
+	meter := &progress.NullProgress{}
+	return snappy.CheckSnap(snapFilePath, flags, meter)
+}
+
+func (s *defaultBackend) SetupSnap(snapFilePath string, flags snappy.InstallFlags) (string, error) {
+	meter := &progress.NullProgress{}
+	return snappy.SetupSnap(snapFilePath, flags, meter)
+}
+
+func (s *defaultBackend) CopySnapData(snapInstPath string, flags snappy.InstallFlags) error {
+	sn, err := snappy.NewInstalledSnap(filepath.Join(snapInstPath, "meta", "snap.yaml"))
+	if err != nil {
+		return err
+	}
+	return snappy.CopyData(sn, flags, &progress.NullProgress{})
+}
+
+func (s *defaultBackend) GenerateSecurityProfile(snapInstPath string) error {
+	sn, err := snappy.NewInstalledSnap(filepath.Join(snapInstPath, "meta", "snap.yaml"))
+	if err != nil {
+		return err
+	}
+	return snappy.GenerateSecurityProfile(sn)
+}
+
+func (s *defaultBackend) FinalizeSnap(snapInstPath string, flags snappy.InstallFlags) error {
+	sn, err := snappy.NewInstalledSnap(filepath.Join(snapInstPath, "meta", "snap.yaml"))
+	if err != nil {
+		return err
+	}
+	return snappy.FinalizeSnap(sn, flags, &progress.NullProgress{})
+}
+
+func (s *defaultBackend) UndoSetupSnap(snapFilePath string) error {
+	meter := &progress.NullProgress{}
+	snappy.UndoSetupSnap(snapFilePath, meter)
+	return nil
```

Comment:
If an `error` is never returned do we still want it in the function signature/interface?